src/libclipper/src/query_processor.cpp

```diff
@@ -101,48 +39,40 @@ std::shared_ptr<StateDB> QueryProcessor::get_state_table() const {
 
 boost::future<Response> QueryProcessor::predict(Query query) {
   long query_id = query_counter_.fetch_add(1);
-  std::vector<PredictTask> tasks;
-  std::string serialized_state;
-
-  if (query.selection_policy_ == "EXP3") {
-    auto tasks_and_state =
-        select_predict_tasks<Exp3Policy>(query, query_id, get_state_table());
-    tasks = tasks_and_state.first;
-    serialized_state = tasks_and_state.second;
-
-  } else if (query.selection_policy_ == "EXP4") {
-    auto tasks_and_state =
-        select_predict_tasks<Exp4Policy>(query, query_id, get_state_table());
-    tasks = tasks_and_state.first;
-    serialized_state = tasks_and_state.second;
-
-  } else if (query.selection_policy_ == "EpsilonGreedy") {
-    auto tasks_and_state = select_predict_tasks<EpsilonGreedyPolicy>(
-        query, query_id, get_state_table());
-    tasks = tasks_and_state.first;
-    serialized_state = tasks_and_state.second;
-
-  } else if (query.selection_policy_ == "UCB") {
-    auto tasks_and_state =
-        select_predict_tasks<UCBPolicy>(query, query_id, get_state_table());
-    tasks = tasks_and_state.first;
-    serialized_state = tasks_and_state.second;
-  } else {
+  boost::future<Response> error_response =
+      boost::make_ready_future(Response{query, query_id, 20000, Output{1.0, {}},
+                                        std::vector<VersionedModelId>()});
+  auto current_policy_iter = selection_policies_.find(query.selection_policy_);
+  if (current_policy_iter == selection_policies_.end()) {
     log_error_formatted(LOGGING_TAG_QUERY_PROCESSOR,
                         "{} is an invalid selection policy",
                         query.selection_policy_);
     // TODO better error handling
-    return boost::make_ready_future(
-        Response{query, query_id, 20000, Output{1.0, {std::make_pair("m1", 1)}},
-                 std::vector<VersionedModelId>()});
+    return error_response;
   }
+  std::shared_ptr<SelectionPolicy> current_policy = current_policy_iter->second;
+
+  auto state_opt = state_db_->get(StateKey{query.label_, query.user_id_, 0});
+  if (!state_opt) {
```

Comment:
After changing the specified selection policy in the end-to-end benchmark from "EXP3" to "DefaultOutputSelectionPolicy", the benchmark is still logging an error here:
```
[04:22:23.560][error] [QUERYPR...] No selection state found for query with label: test
```